tools/ansible/wsu.yaml

```diff
@@ -0,0 +1,135 @@
+---
+- hosts: localhost
+  vars:
+     openshift_node_machineconfigpool: 'worker'
+     openshift_node_bootstrap_port: 22623
+     openshift_node_bootstrap_endpoint: "{{ openshift_node_bootstrap_server }}:{{ openshift_node_bootstrap_port }}/config/{{ openshift_node_machineconfigpool }}"
+    
+  tasks:
+    - name: Validate given input parameters meet the requirement
+      fail:
+          msg: "Please provide all the following parameters to run this playbook [ kubelet_location, kubeovn_location, wmcb_location, openshift_node_bootstrap_server ]"
+      when: kubelet_location is not defined or
+             wmcb_location is not defined or
+             openshift_node_bootstrap_server is not defined
+
+    - name: Show ignition file endpoint 
+      debug:
+        msg: "Ignition file endpoint is {{ openshift_node_bootstrap_endpoint }}"
+
+    - name: Create temporary pod template file
+      file:
+        path: curl_pod.yaml
+        state: touch
+        mode: '0644'
+
+    - name: Copy pod template to temporary pod file
+      copy:
+        src: curl_pod_template.yaml
+        dest: curl_pod.yaml
+        mode: '0644'
+
+    - name: Change endpoint location in temporary pod template entrypoint 
+      replace:
+        path: curl_pod.yaml
+        regexp: "<endpoint>"
+        replace: "{{ openshift_node_bootstrap_endpoint }}" 
+
+    - name: Run pod on Cluster
+      command: oc apply -f curl_pod.yaml
+
+    - name: Wait until pod is created 
+      command: oc get pod curl-ign-pod
+      register: pod_status
+      delay: 5
+      retries: 10
+      until:
+        - '"ContainerCreating" not in pod_status.stdout'
+
+    - name: Create temporary build directory to store files
+      tempfile:
+        state: directory
+      register: build_tmp
+
+    - name: Show temporary path directory
+      debug:
+        msg: "Temporary build path is {{ build_tmp.path }}"
+      when: build_tmp is defined
+
+    - name: Fetch cluster ignition file to controller machine
+      shell: oc logs curl-ign-pod > "{{ build_tmp.path }}/worker.ign"
+      when: build_tmp is defined
+
+    - name: Download Kubernetes archive to controller machine
+      get_url:
+        url: "{{ kubelet_location }}"
+        dest: "{{ build_tmp.path }}/kube.tar.gz"
+        mode: 0400
+      when: build_tmp is defined
+
+    - name: Unarchive Kubenetes binary
+      unarchive:
+        src: "{{ build_tmp.path }}/kube.tar.gz"
+        dest: "{{ build_tmp.path }}" 
+      when: build_tmp is defined
+
+    - name: Download wmcb binary to controller machine
+      get_url:
+        url: "{{ wmcb_location }}"
+        dest: "{{ build_tmp.path }}/wmcb.exe"
+        mode: 0400
+      when: build_tmp is defined
```

Comment:
Do you need to wait until the temp path's permission has been changed successfully?